flutter/integration_test/first_test.dart

```diff
@@ -54,19 +54,12 @@ void checkAccuracy(BenchmarkExportResult benchmarkResult) {
   );
   expectedMap!;
 
-  // backends are not forced to follow backendSettingsInfo values
-  // we should use backendInfo.accelerator, it always represents real accelerator value
-  var accelerator = benchmarkResult.backendInfo.acceleratorName;
-  if (accelerator == 'ACCELERATOR_NAME') {
```

Comment:
Why is this removed?

Reply:
It would make the code easier to follow. It would be more simple to enforce backend to set `accelerator` field when adding accuracy test for it.

Reply:
It makes sense to add a test for it, to make sure that our default config doesn't use accelerator fallback.

Reply:
Maybe it would be better to disable the fallback completely.
Backends should detect supported hardware when choosing the config, and then they should follow the settings, and return an error if something doesn't match.

But this is not related to this PR, we should create a separate issue for this.

Reply:
I think we can update the `expectedMap` to `accelerator|framework`
```dart
  final accelerator = benchmarkResult.backendSettings.acceleratorCode;
  final framework = benchmarkResult.backendSettings.framework;
  final expectedValue = expectedMap['$accelerator|$framework'];
  final tag =
      '${benchmarkResult.benchmarkId} [accelerator: $accelerator | framework: $framework]';
  expect(
    expectedValue,
    isNotNull,
    reason: 'missing expected accuracy for $tag',
  );
```
In that case, a fallback will fail this test since it cannot find the expected accuracy

Reply:
This will work, for our current setup.

There is nothing preventing you from using the same `framework` value for different accelerators.

In this PR the `framework` field does not match its description. I'm kinda OK with this, but if we were to correctly follow the description of this field both `npu` and `cpu` accelerator values would correspond to the same `TFLite` framework description, for example.
I don't think we should rely on such inconsistent behavior.

Reply:
Adding the test that accelerator in config and reported accelerator match is simpler and more reliable.

Reply:
> Adding the test that accelerator in config and reported accelerator match is simpler and more reliable.

Yes, it is better.
Updated.